fv3core/stencils/ytp_v.py

```diff
@@ -1,230 +1,165 @@
+            br = al[0, 1, 0] - v[0, 0, 0]
 
-def zero_br_bl_corners_north(br, bl):
-    grid = spec.grid
-    corner_domain = (1, 2, grid.npz)
-    if grid.nw_corner:
-        br_bl_corner(br, bl, origin=(grid.is_, grid.je, 0), domain=corner_domain)
-    if grid.ne_corner:
-        br_bl_corner(br, bl, origin=(grid.ie + 1, grid.je, 0), domain=corner_domain)
+            # Zero corners
+            with horizontal(
+                region[i_start, j_start - 1 : j_start + 1],
+                region[i_start, j_end : j_end + 2],
+                region[i_end + 1, j_start - 1 : j_start + 1],
+                region[i_end + 1, j_end : j_end + 2],
+            ):
+                bl = 0.0
+                br = 0.0
 
+            flux = get_flux_v(v, c, rdy, bl, br)
 
-def compute(c, u, v, flux):
+        else:
+            dm = yppm.dm_jord8plus(v)
+            al = yppm.al_jord8plus(v, dm)
+
+            assert __INLINED(jord == 8)
+            # {
+            bl, br = yppm.blbr_jord8(v, al, dm)
+            # }
+
+            assert __INLINED(namelist.grid_type < 3)
+            # {
+            with horizontal(region[:, j_start - 1]):
+                bl, br = yppm.south_edge_jord8plus_0(v, dy, dm)
+                # bl, br = yppm.pert_ppm_standard_constraint_fcn(u, bl, br)
+
+            with horizontal(region[:, j_start]):
+                bl, br = yppm.south_edge_jord8plus_1(v, dy, dm)
+                # bl, br = yppm.pert_ppm_standard_constraint_fcn(u, bl, br)
+
+            with horizontal(region[:, j_start + 1]):
+                bl, br = yppm.south_edge_jord8plus_2(v, dm, al)
+                bl, br = yppm.pert_ppm_standard_constraint_fcn(v, bl, br)
+
+            with horizontal(region[:, j_end - 1]):
+                bl, br = yppm.north_edge_jord8plus_0(v, dm, al)
+                bl, br = yppm.pert_ppm_standard_constraint_fcn(v, bl, br)
+
+            with horizontal(region[:, j_end]):
+                bl, br = yppm.north_edge_jord8plus_1(v, dy, dm)
+                # bl, br = yppm.pert_ppm_standard_constraint_fcn(u, bl, br)
+
+            with horizontal(region[:, j_end + 1]):
+                bl, br = yppm.north_edge_jord8plus_2(v, dy, dm)
+                # bl, br = yppm.pert_ppm_standard_constraint_fcn(u, bl, br)
+
+            # Zero corners
+            with horizontal(
+                region[i_start, j_start - 1 : j_start + 1],
+                region[i_start, j_end : j_end + 2],
+                region[i_end + 1, j_start - 1 : j_start + 1],
+                region[i_end + 1, j_end : j_end + 2],
+            ):
+                bl = 0.0
+                br = 0.0
+            # }
+
+            flux = get_flux_v_ord8plus(v, c, rdy, bl, br)
+
+
+def compute(c: FloatField, u: FloatField, v: FloatField, flux: FloatField):
+    """
+    Compute flux of kinetic energy in x-dir using the PPM method.
+
+    Notes:
+        u is passed in here, but is unused in the stencil.
+
+    Args:
+        c (in): Courant number
```

Comment:
Careful, I think this is the courant number "in flux form". Double-check units in how it's used below to confirm this. Also, I suggest renaming this everywhere (in this file) to "courant". No need to rename it where this function is called, we can work our way up over time (and besides, we might need to rename it further).